pkg/node/id.go

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package node
+
+import (
+	"crypto/rand"
+
+	base58 "github.com/jbenet/go-base58"
+)
+
+// ID is the unique identifer of a Node in the overlay network
+type ID string
+
+// String transforms the ID to a string type
+func (n *ID) String() string {
+	return string(*n)
+}
+
+// Bytes transforms the ID to type []byte
+func (n *ID) Bytes() []byte {
+	return []byte(*n)
+}
+
+// StringToID trsansforms a string to a ID
+func StringToID(s string) *ID {
+	n := ID(s)
+	return &n
+}
+
+// NewID returns a pointer to a newly intialized ID
+// TODO@ASK: this should be removed; superseded by `CASetupConfig.Create` / `IdentitySetupConfig.Create`
+func NewID() (*ID, error) {
+	b, err := newID()
+	if err != nil {
+		return nil, err
+	}
+
+	bb := ID(base58.Encode(b))
```

Comment:
i think it's worth separating the value from how it's encoded. my vote is we track ids as binary data, and only worry about encoding/decoding when we're parsing/displaying ids (the String method or the constructor, for instance)

Reply:
agreed -- I didn't just add this code, only moved it from `pkg/kademlia` to pkg/node. I'll fix this. wondering, is there a better place ?

Reply:
oh, let's just keep it another pr